github-api-call-inventory/github-api-call-inventory.ipynb

```diff
@@ -0,0 +1,520 @@
+    "        if isinstance(obj, int) and obj != 0:\n",
+    "            raise TypeError(f\"adding {obj!r} to an {type(self)!r} is not supported\")\n",
+    "        if isinstance(obj, int) and obj == 0:\n",
+    "            obj = Event(0, 0, 0)\n",
+    "        return Event(*[(getattr(self, field.name) + getattr(obj, field.name)) for field in fields(self.__class__)])\n",
+    "    \n",
+    "    __radd__ = __add__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a44d1bba",
+   "metadata": {},
+   "source": [
+    "## Estimating the number of API calls"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "c06631b6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def github_api_calls(\n",
+    "    # The number of chroots configured to be built in Copr\n",
+    "    chroots=3, \n",
+    "    # GitHub server-to-server rate limit per hour\n",
+    "    rate_limit=5000,\n",
+    "    # Calls to load .packit.yaml.\n",
+    "    # The current implementation cycles through a list of 7 files.\n",
+    "    # If the config file is .packit.yaml, this is loaded from 2 calls.\n",
+    "    # If the config file is packit.yaml, it's loaded from 5 calls.\n",
+    "    # The +1 call is to try to find the specfile path.\n",
+    "    load_packit_yaml=2+1,\n",
+    "    # Whether builds are enabled\n",
+    "    builds_enabled=True, \n",
+    "    # Whether tests are enabled\n",
+    "    tests_enabled=True, \n",
+    "    # Load packit.yaml once during the pipeline run\n",
+    "    load_packit_yaml_once=False, \n",
+    "    # Set all flags (including test ones) when the build status changes\n",
+    "    all_flags=True,\n",
+    "    # Check if the repo is private only in the beginning\n",
+    "    is_private_once=False,\n",
+    "):\n",
+    "    build_status_flag = int(builds_enabled)\n",
+    "    test_status_flag = int(tests_enabled)\n",
+    "    events = {\n",
+    "        \"PR is updated\": Event(1, load_packit_yaml, chroots * (build_status_flag + test_status_flag)),\n",
+    "    }\n",
+    "    load_packit_yaml = load_packit_yaml if not load_packit_yaml_once else 0\n",
+    "    flags_to_set = (build_status_flag + test_status_flag) if all_flags else build_status_flag\n",
+    "    is_private = 0 if is_private_once else 1\n",
+    "    event = Event(is_private, load_packit_yaml, flags_to_set)\n",
+    "    events.update({\n",
+    "        \"SRPM build started\": chroots * event,\n",
+    "        \"SRPM build finished\": chroots * event,\n",
+    "        \"RPM build started\": chroots * event,\n",
+    "        \"RPM build finished\": chroots * event,\n",
+    "    })\n",
+    "    if tests_enabled:\n",
+    "        event = Event(is_private, load_packit_yaml, test_status_flag)\n",
+    "        events.update({\n",
+    "        \"Tests started\": chroots * event,\n",
+    "        \"Tests finished\": chroots * event,\n",
+    "    })\n",
```

Comment:
Are there any other events received during a pipeline run?

Reply:
I think "SRPM build in Copr was submitted..." is the only missing

Reply:
Is that a separate event, or that's the status which is set when the update event for the PR is received?

Reply:
it is not a separate event, but it happens in a new task:
we receive a Github PR-related event, we create the "Task was accepted status"  ( + do the config loading, private check,..) and then create a new specific task and there we create the "SRPM build in Copr was submitted..." status

Reply:
Thanks, updated this.